### DISCUSSION TITLE
Welcome to squirrel Discussions!

### DISCUSSION THREAD

**Discussion by hekailiang** [hekailiang/squirrel#123] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by facelezzzz** [comment]
I want to order a special id for a state machine

In AbstractStateMachine.class

`data.write().identifier(configuration.getIdProvider().get());`

but there is no way set it manually.Id provider is only invoked by no-arg method

**Comment by shub124** [comment]
Hi. I have been exploring the functionalities that squirrel provides. Squirrel supports performance monitoring and diagnosis on state machines. Does that mean it is stateful in nature?

**Comment by codeShaurya** [comment]
I want to discuss this issue: https://github.com/hekailiang/squirrel/issues/133

**Comment by sameerargadeinfy** [comment]
Where can we see working examples of parallel state machines, hierarchical state machines , linked machines etcc

**Comment by jfelrod1960** [comment]
I'm developing a real-time strategy game that involves vehicles and each vehcile has a state machine used to control the vehicle's behavior.  Two questions.  One, it cannot load the StaticLoggerBinder class.  How can I stop it from trying?  Two is this still supported if I have issues or would I have to clone it and keep it up myself?  Thanks for your time!
